_devExtensions/editPassword-controller.js

```diff
@@ -0,0 +1,69 @@
+const bcrypt = require('bcryptjs')
+const BS3Database = require('better-sqlite3')
+const path = require('path')
+const passwordValidationSchema = require('./password-validation.schema')
+
+const updateUserPassword = (db, userId, newPassword) => {
+  const hashedPassword = bcrypt.hashSync(newPassword)
+  const now = new Date()
+  const futureTime = new Date(now.getTime() + 60 * 60000)
+  const futureTimeString = futureTime.toISOString()
+  const query = `
+        UPDATE _users
+        SET hashed_password = ?, lastUpdatedAt = ?, updateBefore = ?
+        WHERE id = ?;
+      `
+  db.prepare(query).run(
+    hashedPassword,
+    new Date().toISOString(),
+    futureTimeString,
+    userId
+  )
+}
+
+const editPasswordController = async (req, res) => {
+  let mainDb
+
+  try {
+    mainDb = new BS3Database(path.join(process.cwd(), 'db/RCO2.sqlite'))
+    const { fields: queryFields } = req.body
+    queryFields.createdAt = new Date().toISOString()
+    const { userId, newPassword } = queryFields
+
+    await passwordValidationSchema.validate(newPassword)
+
+    queryFields.newPassword = bcrypt.hashSync(newPassword)
+
+    const fields = Object.fromEntries(
+      Object.entries(queryFields).filter(([_, value]) => value !== null)
+    )
+    const fieldsString = Object.keys(fields).join(', ')
+    const valuesString = Object.values(fields)
+      .map((value) => {
+        if (typeof value === 'string') {
+          return `'${value}'`
+        }
+        return value
+      })
+      .join(', ')
+
+    let values = `(${fieldsString}) VALUES (${valuesString})`
+    if (valuesString === '') {
+      values = 'DEFAULT VALUES'
+    }
+
```

Comment:
I'm pretty sure we can modify the above code to use a const for values, something like:
```
   const values = valuesString === '' ?  'DEFAULT VALUES' : [longer-string-definition] 
```

Reply:
Hold on.... We don't use the value of the field `values`